### ISSUE TITLE
GetNextObjectInArea bug

### ISSUE THREAD

**Issue by Shad000w** [Beamdog/nwn-issues#84]
**To Reproduce**

- Load attached save or module.
- Use the crystall ball placeable.
- Check log - in log you will see two GetFirst/NextObjectInArea loops, one will report 4 valid objects, the next will report 5 valid object and one object, stinking bug, twice.
[000872 - reprobug15.zip](https://github.com/Beamdog/nwn-issues/files/4743051/000872.-.reprobug15.zip)
[repromodule15.zip](https://github.com/Beamdog/nwn-issues/files/4743053/repromodule15.zip)

**Specifics**

 - OS , tested only on windows, both client and server
 - Game build: verified in 1.69, 1.77 and 1.80 (8193.13)
 - Distro platform: Beamdog and Steam
 - none

**If needed, describe the bug**

The script in the module performing the loops is this:

```
#include "70_inc_spells"

void ListAllObjects2()
{
    SendMessageToPC(GetFirstPC(),"listing all objects2");
    object oArea = GetArea(OBJECT_SELF);
    object oObject = GetFirstObjectInArea(oArea);
    effect eDam;
    eDam = EffectDamage(8, DAMAGE_TYPE_SONIC);
    SendMessageToPC(GetFirstPC(),"first object: "+GetName(oObject)+", id: "+ObjectToString(oObject));
    while(oObject != OBJECT_INVALID)
    {
        oObject = GetNextObjectInArea(oArea);
    ApplyEffectToObject(DURATION_TYPE_INSTANT,eDam,oObject);
        SendMessageToPC(GetFirstPC(),"next object: "+GetName(oObject)+", id: "+ObjectToString(oObject));
    }
}

void ListAllObjects()
{
    SendMessageToPC(GetFirstPC(),"listing all objects");
    object oArea = GetArea(OBJECT_SELF);
    object oObject = GetFirstObjectInArea(oArea);
    SendMessageToPC(GetFirstPC(),"first object: "+GetName(oObject)+", id: "+ObjectToString(oObject));
    while(oObject != OBJECT_INVALID)
    {
        oObject = GetNextObjectInArea(oArea);
        SendMessageToPC(GetFirstPC(),"next object: "+GetName(oObject)+", id: "+ObjectToString(oObject));
    }
}

void main()
{
ListAllObjects();
AssignCommand(GetFirstPC(),ListAllObjects2());
//    object oTarget = FIX_GetFirstObjectInShape(SHAPE_SPHERE, 5.0, GetLocation(OBJECT_SELF), TRUE, OBJECT_TYPE_CREATURE | OBJECT_TYPE_DOOR | OBJECT_TYPE_PLACEABLE);
//    oTarget = FIX_GetNextObjectInShape(SHAPE_SPHERE, 5.0, GetLocation(OBJECT_SELF), TRUE, OBJECT_TYPE_CREATURE | OBJECT_TYPE_DOOR | OBJECT_TYPE_PLACEABLE);
}
```
(please ignore the include the commented lines)

Basically it looks like that the object in area pointer gets messed up if there is a damage effect applied by player to some npc at the same time. In this case it listed one creature twice, in another case (which lead to this testing) it was returning OBJECT_INVALID instead of next object.

This didn't happen when the damage was applied by placeable to player.

I've seen similar issue with another functions like GetFirst/NextObjectInShape several years ago. I will try if I can reproduce it.

**Comment by BhaalM** [comment]
I can confirm this happens also with GetFirst/NextObjectInShape but I've not been able to reproduce it.

**Comment by Soren217** [comment]
I was unable to reproduce this - could you check if it still occurs in the latest dev version, and reopen if so? Thanks.
